### PR TITLE
FED-3281 Update rename codemod

### PR DESCRIPTION
**Pull request by sydneyjodon-wk**
## Motivation
  
Make some last updates to prep the rename codemod for the sourcegraph batch
## Changes
  
- Update the codemod to:
  - no longer make changes to import namespaces
  - update `data_grid_premium.dart` entrypoint which is different on `unify_ui`
  - update / add fixme comments to make sure everything that needs to be manually QAed has a comment saying so
  - add `LinkButton` constants to list to be updated
- Update tests
## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
      - [ ] Run updated tests locally since they aren't run in CI
      - [ ] CI passes
      - [ ] Run the codemod on another repo (there are a few edge cases that the codemod won't cover that we'll just have to update on a case-by-case basis)
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Frontend Frameworks Design member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/unify_package_rename_suggestors/import_renamer.dart [hunk 1]
```diff
@@ -52,29 +50,18 @@ Suggestor importRenamerSuggestorBuilder({
...
-          newNamespace = specialCaseNamespace;
-        }
       }
 
       if (newImportUri != null) {
         // Collect info on new imports to add.
         newImportsInfo.add(UnifyImportInfo(newImportUri,
-            namespace: newNamespace,
+            namespace: namespace,
             showHideInfo: import.combinators
```
Comment (greglittlefield-wf) [inline lib/src/unify_package_rename_suggestors/import_renamer.dart:65]: #nit Not directly related to the changes in this PR, but while QAing I noticed that there's an extra space inserted between the end of the imports before their semicolons. For example
```diff
-import 'package:react_material_ui/react_material_ui.dart';
+import 'package:unify_ui/unify_ui.dart' ;
```

Those changes end up getting formatted out, if the formatter actually runs on all files...

It looks like it's due to the `importInfo.showHideInfo != null ? ' ${importInfo.showHideInfo}' : ''` below combined with this evaluating to an empty string if `import.combinators` is empty. 

I made this change locally, which fixed the issue!
```diff
diff --git a/lib/src/unify_package_rename_suggestors/import_renamer.dart b/lib/src/unify_package_rename_suggestors/import_renamer.dart
index c1cd79a..228660d 100644
--- a/lib/src/unify_package_rename_suggestors/import_renamer.dart
+++ b/lib/src/unify_package_rename_suggestors/import_renamer.dart
@@ -65,4 +65,3 @@ Suggestor importRenamerSuggestorBuilder({
-            showHideInfo: import.combinators
-                .map((c) => c.toSource())
-                .toList()
-                .join(' ')));
+            showHideInfo: import.combinators.isEmpty
+                ? null
+                : import.combinators.map((c) => c.toSource()).join(' ')));
```

Would you mind folding that change into this PR? I think it'd be worth it to reduce make formatting less of a crucial step to getting a good diff.

**Review by greglittlefield-wf — CHANGES_REQUESTED** [review]
One question and a #nit, otherwise LGTM, and works and has passing tests locally!

File: test/unify_package_rename_suggestors/unify_rename_suggestor_test.dart [hunk 2]
```diff
@@ -261,20 +91,30 @@ void main() {
...
       unify_wsd.WsdAlert()();
+      // FIXME(unify_package_rename) Check what theme provider is wrapping this component: if it is a UnifyThemeProvider, update this to `Alert` from `unify_ui/components/alert.dart`, manually QA this component, and remove this FIXME; otherwise, remove this FIXME.
       unify_wsd.WsdAlert();
+      // FIXME(unify_package_rename) Check what theme provider is wrapping this component: if it is a UnifyThemeProvider, update this to `Alert` from `unify_ui/components/alert.dart`, manually QA this component, and remove this FIXME; otherwise, remove this FIXME.
       unify_wsd.WsdAlert;
+      // FIXME(unify_package_rename) Check what theme provider is wrapping this component: if it is a UnifyThemeProvider, update this to `AlertPropsMixin` from `unify_ui/components/alert.dart`, manually QA this component, and remove this FIXME; otherwise, remove this FIXME.
+      unify_wsd.WsdAlertPropsMixin;
       unify_wsd.WsdAlertSize.small;
```
[the remark this answers is not in the document]
Comment (sydneyjodon-wk) [reply test/unify_package_rename_suggestors/unify_rename_suggestor_test.dart:102]: Yeah I debated removing the namespace, but then was worried about it being too easy to import these, but you're right everything in that entrypoint has Wsd in the name so probably unnecessary! I'll remove it!

File: lib/src/unify_package_rename_suggestors/import_renamer.dart (hunk 1 above)
Comment (sydneyjodon-wk) [reply lib/src/unify_package_rename_suggestors/import_renamer.dart:65]: Nice! Thank you! Yeah I noticed that in my sourcegraph run and wasn't sure if it was worth fixing so thanks for finding the issue!!

File: test/unify_package_rename_suggestors/unify_rename_suggestor_test.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (sydneyjodon-wk) [reply test/unify_package_rename_suggestors/unify_rename_suggestor_test.dart:102]: Uh actually it might be hard to remove it because the import adder [looks for errors related to the namespace](https://github.com/Workiva/over_react_codemod/blob/8b4e889f148c5fec8f31ef81f6087fb61665ab2f/lib/src/util/importer.dart#L50) - do you know of an easy way to work around this?

[the remark this answers is not in the document]
Comment (greglittlefield-wf) [reply test/unify_package_rename_suggestors/unify_rename_suggestor_test.dart:102]: Gah, funn. Hmm I think you'd have to manually yield a patch for that import.

Could use the same logic that that importer uses, something like:
```diff
 Future<void> generatePatches() async {
     ...
     result.unit.visitChildren(this);
+
+    if (needsWsdImport) {
+      final insertInfo = insertionLocationForPackageImport(
+          unifyWsdUri, result.unit, result.lineInfo);
+      yieldPatch(
+          insertInfo.leadingNewlines +
+              "import '$unifyWsdUri';" +
+              insertInfo.trailingNewlines,
+          insertInfo.offset,
+          insertInfo.offset);
+    }
   }
```

Not too bad, but also not a trivial change, so maybe not worth it after all; I could go either way

[the remark this answers is not in the document]
Comment (sydneyjodon-wk) [reply test/unify_package_rename_suggestors/unify_rename_suggestor_test.dart:102]: oo I like that! That also fixes a random edge case I found where there's more than one error and it doesn't see the undefined identifier error and doesn't add the import! 🎉

**Review by greglittlefield-wf — APPROVED** [review]
- Changes look good
- Tests pass locally
- Output looks good when run against packages locally

+10

**Comment by greglittlefield-wf** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM
